src/lib.rs

```diff
@@ -133,32 +105,44 @@ impl TimsReader {
         self.reader
             .read_all_spectra()
             .iter()
-            .map(|x| PySpectrum::new(x))
+            .map(PySpectrum::new)
             .collect()
     }
 
     fn __repr__(slf: &PyCell<Self>) -> PyResult<String> {
         let class_name: &str = slf.get_type().name()?;
         Ok(format!(
-            "{}(path={})",
+            "{}(path='{}')",
             class_name,
             slf.borrow().path.clone()
         ))
     }
 
-    fn resolve_mzs(slf: &PyCell<Self>, tofs: Vec<u32>) -> Vec<f64> {
-        let converter = &slf.borrow().tof_converter;
-        tofs.iter().map(|x| converter.convert(*x)).collect()
+    fn resolve_mzs(slf: &PyCell<Self>, tofs: Vec<u32>) -> PyResult<Vec<f64>> {
+        match &slf.borrow().reader.get_frame_converter() {
```

Comment:
This should be get_tof_converter ... the frame converter converts frames to RTs. This bug made me grow gray hair :P

Reply:
Whoops. Let's blame copilot. And myself for not verifying copilot 🙄